File: Delegate/backend/src/services/email/EmailProviderService.ts
import type { OOOSetting } from '../../types/oooSetting';
import type { Delegation } from '../../types/delegation';

export interface EmailProviderService {
  // OOO Settings
  setOOOSetting(userId: string, setting: OOOSetting): Promise<void>;
  clearOOOSetting(userId: string): Promise<void>;
  getOOOSetting(userId: string): Promise<OOOSetting | null>;

  // Delegation
  createDelegation(delegation: Delegation): Promise<void>;
  removeDelegation(delegation: Delegation): Promise<void>;
  updateDelegation(delegation: Delegation): Promise<void>;
  getDelegations(userId: string): Promise<Delegation[]>;
}

export enum EmailProviderType {
  OUTLOOK_365 = 'outlook365',
  GMAIL = 'gmail',
  // Add more providers as needed
} 

File: Delegate/backend/src/main/java/com/delegate/service/email/EmailProviderService.java
package com.delegate.service.email;

import com.delegate.model.OOOSetting;
import com.delegate.model.Delegation;
import java.util.List;

public interface EmailProviderService {
    // OOO Settings
    void setOOOSetting(String userId, OOOSetting setting);
    void clearOOOSetting(String userId);
    OOOSetting getOOOSetting(String userId);

    // Delegation
    void createDelegation(Delegation delegation);
    void removeDelegation(Delegation delegation);
    void updateDelegation(Delegation delegation);
    List<Delegation> getDelegations(String userId);
} 